dts/bindings/posix/zephyr,posix.yaml

```diff
@@ -0,0 +1,20 @@
+#
```

Comment:
we typically haven't had yaml for such things, is this actually needed for anythingn?

Reply:
No it is not needed, file is removed

Reply:
Update: looks like it is needed for the bsim board else dts extraction script will fail because not a single binding is found.

https://github.com/zephyrproject-rtos/zephyr/blob/57629724c2b28acd857747cd232c5a975fe55591/scripts/dts/extract_dts_includes.py#L377

Reply:
Removed DTS support for nrf52 bsim support as such the unnecessary bindings file could be removed